### PR TITLE
5014 known slack filtering

### PR DESCRIPTION
**Pull request by esaunders**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/datamodel/DataSourcesKnownAndSlackFilter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    static {
+        /**
+         * Watch for user preference changes and update variables inherited from
+         * our parent. The actual filtering is provided by our parent class.
+         */
+        UserPreferences.addChangeListener((PreferenceChangeEvent evt) -> {
+            if (evt.getKey().equals(UserPreferences.HIDE_KNOWN_FILES_IN_DATA_SRCS_TREE)) {
+                filterKnown = UserPreferences.hideKnownFilesInDataSourcesTree();
+            } else if (evt.getKey().equals(UserPreferences.HIDE_KNOWN_FILES_IN_DATA_SRCS_TREE)) {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/DataSourcesKnownAndSlackFilter.java]: Looks like a copy-paste error here.

File: Core/src/org/sleuthkit/autopsy/datamodel/ViewsKnownAndSlackFilter.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    static {
+        /**
+         * Watch for user preference changes and update variables inherited from
+         * our parent. The actual filtering is provided by our parent class.
+         */
+        UserPreferences.addChangeListener((PreferenceChangeEvent evt) -> {
+            if (evt.getKey().equals(UserPreferences.HIDE_KNOWN_FILES_IN_VIEWS_TREE)) {
+                filterKnown = UserPreferences.hideKnownFilesInViewsTree();
+            } else if (evt.getKey().equals(UserPreferences.HIDE_KNOWN_FILES_IN_VIEWS_TREE)) {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/ViewsKnownAndSlackFilter.java]: Same thing.

File: Core/src/org/sleuthkit/autopsy/datamodel/DataSourcesKnownAndSlackFilter.java (hunk 1 above)
Comment (esaunders) [reply Core/src/org/sleuthkit/autopsy/datamodel/DataSourcesKnownAndSlackFilter.java]: Fixed

File: Core/src/org/sleuthkit/autopsy/datamodel/ViewsKnownAndSlackFilter.java (hunk 2 above)
Comment (esaunders) [reply Core/src/org/sleuthkit/autopsy/datamodel/ViewsKnownAndSlackFilter.java]: Fixed

**Pull request merged**